server/src/main/java/com/cloud/storage/StoragePoolAutomationImpl.java

```diff
@@ -346,6 +346,11 @@ public boolean cancelMaintain(DataStore store) {
                 if (logger.isDebugEnabled()) {
                     logger.debug("ModifyStoragePool add succeeded");
                 }
+                try {
+                    storageManager.connectHostToSharedPool(host, pool.getId());
+                } catch (Exception e) {
+                    logger.warn("Unable to establish a connection between {} and {}", host, pool, e);
+                }
```

Comment:
what if the third of six hosts fails? what would the state of the overall system be at that point?

Reply:
I do not understand your question.
This is called only when `ModifyStoragePool` succeeds, so host should be able to connect to pools.